src/components/Toaster/types.ts

```diff
@@ -18,6 +18,8 @@ export type ToastAction = {
     removeAfterClick?: boolean;
 };
 
+export type RenderIconFunction = (toastProps: ToastProps) => React.ReactNode;
```

Comment:
Let's remove this type completely, and use function notation: `() => React.ReactNode`

Reply:
Do you really want me to remove the arguments?
I added them to give the user the ability to naturally change the icon depending on the type and (possibly) other properties of the toast.
We don't have a toast object.
I doubt a closure will work here, since all the user has is an immutable object that he pass into the add function.
When changing the properties of the toast through the change method, the old closure will no longer be relevant. He will have to change the render function too.

If you don't like the argument, maybe it will be better to have just Icon ReactNode prop (not a function)?